raiden-ts/src/transfers/reducer.ts

```diff
@@ -73,22 +68,18 @@ function transferSignedReducer(state: RaidenState, action: transferSigned): Raid
 
   // transferSigned must be the first action, to init TransferState state
   if (secrethash in state[action.meta.direction]) return state;
-  const channelPath = ['channels', transfer.token_network_address, partner];
-  let channel: Channel | undefined = get(channelPath, state);
+  const key = channelKey({ tokenNetwork: transfer.token_network_address, partner });
+  let channel = state.channels[key];
   if (!channel) return state;
 
-  const locks = [...(channel[end].locks ?? []), lock]; // append lock
+  const locks = [...channel[end].locks, lock]; // append lock
   const locksroot = getLocksroot(locks);
   if (
     transfer.locksroot !== locksroot ||
     // nonce must be next
-    !transfer.nonce.eq(
-      (channel[end].balanceProof ? channel[end].balanceProof!.nonce : Zero).add(1),
-    ) ||
-    !transfer.transferred_amount.eq(
-      channel[end].balanceProof ? channel[end].balanceProof!.transferredAmount : Zero,
-    ) ||
-    !transfer.locked_amount.eq((channel[end].balanceProof?.lockedAmount ?? Zero).add(lock.amount))
+    !transfer.nonce.eq(channel[end].balanceProof.nonce.add(1)) ||
```

Comment:
nitpick: again it would be nice if we could extract the `channel[end].balanceProof` to a variable